### PR TITLE
Manage notification templates

### PR DESCRIPTION
**Pull request by somindatommy**
### Proposed changes in this pull request

- Introducing a new service interface for notification template management.

Related issues: https://github.com/wso2/product-is/issues/6669

### When should this PR be merged

This PR should merge after merging PR: https://github.com/wso2/carbon-identity-framework/pull/2615

### Follow up actions

Following PR should merge after merging this PR.
- https://github.com/wso2-extensions/identity-event-handler-notification/pull/110

### CODE REVIEW

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.identity.governance.exceptions.notiification;
+
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.governance.IdentityMgtConstants;
+
+public class NotificationTemplateManagerClientException extends NotificationTemplateManagerException{
```
Comment (somindatommy) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java]: Formatting error

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtConstants.java [hunk 2]
```diff
@@ -25,6 +25,15 @@ private IdentityMgtConstants(){}
 
     public static final String USER_IDENTITY_CLAIMS = "UserIdentityClaims";
 
+    /**
+     * Class that contains the error scenarios.
+     */
+    public class Error_Scenario{
```
Comment (malithie) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtConstants.java]: formatting

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationChannelManagerServerException.java [hunk 3]
```diff
@@ -78,7 +82,8 @@ private String getDefaultErrorCode() {
 
         String errorCode = super.getErrorCode();
         if (StringUtils.isEmpty(errorCode)) {
-            errorCode = IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_SERVER_ERROR.getCode();
+            errorCode = IdentityMgtConstants.Error_Scenario.NOTIFICATION_CHANNEL_MANAGER
+                    +IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_SERVER_ERROR.getCode();
```
Comment (malithie) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationChannelManagerServerException.java]: formatting

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+     * Get the default error code of the exception.
+     *
+     * @return Error description
+     */
+    private String getDefaultErrorCode() {
+
+        String errorCode = super.getErrorCode();
+        if (StringUtils.isEmpty(errorCode)) {
+            errorCode = IdentityMgtConstants.Error_Scenario.NOTIFICATION_TEMPLATE_MANAGER
+                    +IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_BAD_REQUEST.getCode();
```
Comment (malithie) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java]: formatting

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerException.java [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+     * Get the default error code of the exception.
+     *
+     * @return Error description
+     */
+    private String getDefaultErrorCode() {
+
+        String errorCode = super.getErrorCode();
+        if (StringUtils.isEmpty(errorCode)) {
+            errorCode = IdentityMgtConstants.Error_Scenario.NOTIFICATION_TEMPLATE_MANAGER
+                    +IdentityMgtConstants.ErrorMessages.ERROR_CODE_DEFAULT_UNEXPECTED_ERROR.getCode();
```
Comment (malithie) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerException.java]: formatting

File: pom.xml [hunk 6]
```diff
@@ -621,7 +621,7 @@
         <carbon.kernel.registry.imp.pkg.version.range>[1.0.1, 2.0.0)</carbon.kernel.registry.imp.pkg.version.range>
 
         <!--Carbon Identity Framework Version-->
-        <carbon.identity.framework.version>5.15.28</carbon.identity.framework.version>
+        <carbon.identity.framework.version>5.15.41-SNAPSHOT</carbon.identity.framework.version>
```
Comment (malithie) [inline pom.xml]: We should not upgrade to snapshot versions. Let's get the framework PR merged and update dependency version accordingly and then get this merged

Comment (somindatommy) [reply pom.xml]: +1. I was waiting until the framework is released and then was going to update the version.

Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerException.java (hunk 5 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerException.java]: Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java (hunk 4 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java]: Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationChannelManagerServerException.java (hunk 3 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationChannelManagerServerException.java]: Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtConstants.java (hunk 2 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/IdentityMgtConstants.java]: Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java (hunk 1 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/exceptions/notiification/NotificationTemplateManagerClientException.java]: Fixed: c4a8a4d9f812fa524742115852e7200ab9f521df

**Pull request merged**